### PR TITLE
feat(hyd): aerodynamic forces affecting gear system

### PR DESCRIPTION
**Pull request by crocket63**
Fixes #[issue_no]

## Summary of Changes

This PR adds aerodynamic forces on gear:

- Doors are slightly angled so headwing adds a small opening force.
- Nose gear is pushed back to downlock by headwind, and will have more work to do to retract at highest speeds.
- Main gears have a door panel attached helping them to downlock with headwind

Furthermore, gear system is tuned so gravity extension on ground and in the air have realistic durations

## Additional context

Discord username (if different from GitHub):

## Testing instructions

Only real world data available gives approximately 20+ seconds for full doors opening on ground in gravity extension mode (without wind obviously)
Main gears extend in 20+ seconds too, and nose extends in about 10-13s, not much slower than using hydraulics.

Global gear system check in various conditions:

- Gravity extension on ground, check doors positions (simvars) depending on wind force and direction

- Low speed extension/retraction -> note time it takes for everything to move
- 250kts extension retraction -> note time it takes for everything to move. Timings might be a bit faster but not that much

- Low speed gravity extension -> note time it takes for everything to move. Nose should not be too slow from normal mode, main should be slower
- 250kts gravity extension -> note time it takes for everything to move. Timings might be faster for main and nose
- 350kts -> check if anything stupid happens

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,9 @@
 1. [FMGC] Allow stringing of STARs with non-runway approaches - @tracernz (Mike)
 1. [SD] Improve COND page visuals, fix some visual bugs on the SD - @lukecologne (luke)
 1. [HYD] Aerodynamic forces applied on gear system - @Crocket63 (crocket)
+1. [FMGC] Fix unnatural turn direction for path capture transition - @tracernz (Mike)
+1. [FMGC] Fix unnatural turn direction for course capture transition - @tracernz (Mike)
+1. [FMGC] Update spoiler CD for MSFS SDK clarification - @donstim (donbikes#4084)
```
Comment (tracernz) [inline .github/CHANGELOG.md:51]:
```suggestion
1. [SD] Improve COND page visuals, fix some visual bugs on the SD - @lukecologne (luke)
1. [HYD] Aerodynamic forces applied on gear system - @Crocket63 (crocket)
1. [FMGC] Fix unnatural turn direction for path capture transition - @tracernz (Mike)
1. [FMGC] Fix unnatural turn direction for course capture transition - @tracernz (Mike)
1. [FMGC] Update spoiler CD for MSFS SDK clarification - @donstim (donbikes#4084)

```
Remove space above.

File: src/systems/a320_systems/src/hydraulic/mod.rs [hunk 2]
```diff
@@ -713,6 +713,38 @@ impl A320RudderFactory {
 
 struct A320GearDoorFactory {}
 impl A320GearDoorFactory {
+    fn a320_nose_gear_door_aerodynamics() -> AerodynamicModel {
+        // Faking the single door by only considering right door aero.
+        // Will work with headwind, if testing massige cross wind it will look strange
```
Comment (davidwalschots) [inline src/systems/a320_systems/src/hydraulic/mod.rs:718]:
```suggestion
        // Faking the single door by only considering right door aerodynamics.
        // Will work with headwind, but will cause strange behaviour with massive crosswind.
```

File: src/systems/systems/src/hydraulic/aerodynamic_model.rs [hunk 3]
```diff
@@ -641,6 +641,194 @@ mod tests {
...
+            test_bed(TestAircraft::new(nose_door_body, nose_door_aero_model))
+                .with_wind_speed_and_aoa(Velocity::new::<knot>(200.), Angle::new::<degree>(10.))
+                .with_left_wind(Velocity::new::<meter_per_second>(0.));
+
+        test_bed_nose_door.rotate_body(Angle::new::<degree>(80.));
+
+        test_bed_nose_door.run_without_delta();
+
+        let lateral_force = test_bed_nose_door.query(|a| a.body_aero_force_right_value());
+        let _up_force = test_bed_nose_door.query(|a| a.body_aero_force_up_value());
```
Comment (davidwalschots) [inline src/systems/systems/src/hydraulic/aerodynamic_model.rs:680]:
```suggestion
```

Comment (crocket63) [reply src/systems/systems/src/hydraulic/aerodynamic_model.rs:680]: Welcome back :D   Thanks :)

**Review by davidwalschots — APPROVED** [review]

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #7358
Tier of Testing  : 1
Date             : 04/08/2022

Testing Process: 
Gravity extension on ground, check doors positions (simvars) depending on wind force and direction - **OK**

Low speed extension/retraction -> 17s/15s

250kts extension retraction -> 16s/15s

Low speed gravity extension -> 22s, **OK**

250kts gravity extension -> 18s, **OK**

350kts -> **OK**

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
LGTM

**Pull request merged**